enterprise/dev/ci/internal/ci/operations.go

```diff
@@ -21,59 +21,78 @@ import (
 // arguments to create variations of specific Operations (e.g. with different arguments).
 type Operation func(*bk.Pipeline)
 
-// CoreTestOperations is a core set of tests that should be run in most CI cases. These
-// steps should generally be quite fast.
-func CoreTestOperations(changedFiles ChangedFiles, buildOptions bk.BuildOptions) []Operation {
-	// Default set
+// CoreTestOperationsOptions should be used ONLY to adjust the behaviour of specific steps,
+// e.g. by adding flags, and not as a condition for adding steps or commands.
+type CoreTestOperationsOptions struct {
+	// for clientChromaticTests
+	ChromaticShouldAutoAccept bool
+}
+
+// CoreTestOperations is a core set of tests that should be run in most CI cases. More
+// notably, this is what is used to define operations that run on PRs. Please read the
+// following notes:
+//
+// Operations should ONLY be ADDITIVE based on changedFiles. Please do not remove steps
+// after they are added. changedFiles can be nil to run all tests.
+//
+// opts should be used ONLY to adjust the behaviour of specific steps, e.g. by adding flags,
+// and not as a condition for adding steps or commands.
+//
+// If the conditions for the addition of an operation cannot be expressed using the above
+// arguments, please add it from the switch case within `GeneratePipeline` instead.
+func CoreTestOperations(changedFiles ChangedFiles, opts CoreTestOperationsOptions) []Operation {
+	// Special-case branches provide a nil changedFiles to only run all checks.
```

Comment:
I can't make sense of that comment :smile: Can you rephase it? Is this about `main-dry-run` branches?

Reply:
`main-dry-run`, `image-patch`, etc  - is this a bit clearer?

```suggestion
	// Various runTypes can provide a nil changedFiles to run all checks.
```